src/engine/BMInterfaceGame.php

```diff
@@ -228,6 +230,107 @@ protected function set_random_button_flags($gameId, array $buttonNameArray) {
         }
     }
 
+    /**
+     * Set custom recipes
+     *
+     * @param int $gameId
+     * @param array $buttonNameArray
+     * @param array $customRecipeArray
+     */
+    protected function set_custom_recipes(
+        $gameId,
+        array $buttonNameArray,
+        array $customRecipeArray
+    ) {
+        if (!in_array('CustomBM', $buttonNameArray)) {
+            return TRUE;
+        }
+
+        if ($this->get_config('site_type') != 'development') {
+            $this->set_message('Custom recipes can only be used on development sites.');
+            return FALSE;
+        }
```

Comment:
Sorry, i wanted to resolve this question: is what we want A. CustomBM only on dev, or B. CustomBM on dev or staging?

The argument for "dev or staging" is that people can use staging to test one-off recipes, and it gets me out of the business of having to maintain a third site named "customrecipe" that has some arbitrary version of the code and isn't part of my normal workflow to update/maintain.  (Obviously we'd still stand up a one-off dev site if someone wanted to do long-term testing of a new set, because in that case it's appealing to be able to see the art in-line with the recipes and stuff.  But right now if some tester says "hey, i'd like to try out this recipe for my fanatic", they go to customrecipe.dev, and that's kind of no good because heck knows what code version is on that thing.)

Reply:
Sounds good to me.

Reply:
Oh, to be clear, I believe that once we accept this pull request, we should immediately declare a pull request freeze and proceed on the route to doing a deploy, since the changes are big enough that we want testers to look at staging (non-Custom-BM) behaviour.

Reply:
Works for me.